misc/python/materialize/checks/all_checks/materialized_views.py

```diff
@@ -181,3 +181,76 @@ def validate(self) -> Testdrive:
            """
             )
         )
+
+
+class MaterializedViewsRefresh(Check):
+    def _can_run(self, e: Executor) -> bool:
+        return self.base_version >= MzVersion.parse_mz("v0.82.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > CREATE TABLE refresh_table (x INT);
+                > INSERT INTO refresh_table VALUES (1);
+                > CREATE MATERIALIZED VIEW refresh_view_2s_1 WITH (REFRESH EVERY '2 seconds') AS SELECT DISTINCT(x) FROM refresh_table;
+                > CREATE MATERIALIZED VIEW refresh_view_at_1 WITH (REFRESH AT mz_now()::string::int8) AS SELECT DISTINCT(x) FROM refresh_table;
+            """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > INSERT INTO refresh_table VALUES (2);
+                > CREATE MATERIALIZED VIEW refresh_view_2s_2 WITH (REFRESH EVERY '2 seconds') AS SELECT DISTINCT(x) FROM refresh_table;
+                > CREATE MATERIALIZED VIEW refresh_view_at_2 WITH (REFRESH AT mz_now()::string::int8) AS SELECT DISTINCT(x) FROM refresh_table;
+                """,
+                """
+                > INSERT INTO refresh_table VALUES (3);
+                > CREATE MATERIALIZED VIEW refresh_view_2s_3 WITH (REFRESH EVERY '2 seconds') AS SELECT DISTINCT(x) FROM refresh_table;
+                > CREATE MATERIALIZED VIEW refresh_view_at_3 WITH (REFRESH AT mz_now()::string::int8) AS SELECT DISTINCT(x) FROM refresh_table;
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
```

Comment:
SHOW CREATE MATERIALIZED VIEW is needed here in order to confirm that the `REFRESH AT` option has been preserved across restarts/upgrades

Reply:
Also consider having a MV with a `REFRESH AT` option far in the future and then a check to confirm that the view was never refreshed.

Reply:
Thanks, that found a bug! https://github.com/MaterializeInc/materialize/issues/24244